fiddler/tests/conftest.py

```diff
@@ -1,15 +1,31 @@
+import os
+
 import pytest
 
+FIDDLER_API_KEY = os.environ.get('FIDDLER_API_KEY')
+
 
 @pytest.fixture(scope='session')
-def dd_environment():
-    yield
+def dd_environment(instance, e2e_instance):
+    if FIDDLER_API_KEY:
+        yield e2e_instance
+    else:
+        yield instance
```

Comment:
The instance used will depend on whether the `FIDDLER_API_KEY` environment variable is used or not.

Reply:
Looks good.